Update readme, remove test files from release build

vcanales opened the pull request:
Removes reference to font management still present in the readme file.
Also removes test-related files from release builds.

Commit:
Update readme, removing reference to fonts

mikachan reviewed (commented):
Good spot! I think there are a few other remaining references to fonts too:

readme.txt:
- line 27
- line 30 - mention of fonts

readme.md:
- lines 13 & 14
- lines from 56 - 65 ("Embed Google Fonts or local font files")
- lines 76 & 77
- line 79 - this section mentions editing fonts via the plugin

Commit:
Remove from readme.md

vcanales commented:
> Good spot! I think there are a few other remaining references to fonts too:

Thanks! I've removed those as well. I added a deprecation notice on the readme.md file as well.

Commit:
Remove test-related files from release build

mikachan commented on README.md:
```diff
@@ -55,30 +53,21 @@ Saves user's changes as a [style variation](https://developer.wordpress.org/them
 To use the latest release of the Create Block Theme plugin on your WordPress site: install from the plugins page in wp-admin, or [download from the WordPress.org plugins repository](https://wordpress.org/plugins/create-block-theme).
 
-In the WordPress Admin Dashboard, under Appearance there will be three new pages called:
+In the WordPress Admin Dashboard, under Appearance there will be a new pages called:
```
```suggestion
In the WordPress Admin Dashboard, under Appearance there will be a new page called:
```

mikachan commented on README.md:
```diff
@@ -55,30 +53,21 @@ Saves user's changes as a [style variation](https://developer.wordpress.org/them
--   The font face definitions to the `theme.json` file.
-
-You can continue using your modified theme or you can export it as a new theme containing the new fonts.
+**DEPRECATED:** Font can now be managed using WordPress' native Font Library. [More information](https://wordpress.org/documentation/wordpress-version/version-6-5/#add-and-manage-fonts-across-your-site).
```
```suggestion
**DEPRECATED:** Fonts can now be managed using WordPress' native Font Library. [More information](https://wordpress.org/documentation/wordpress-version/version-6-5/#add-and-manage-fonts-across-your-site).
```

vcanales committed:
Update README.md

Co-authored-by: Sarah Norris <1645628+mikachan@users.noreply.github.com>

vcanales committed:
Update README.md

Co-authored-by: Sarah Norris <1645628+mikachan@users.noreply.github.com>

mikachan commented on README.md:
```diff
@@ -55,30 +53,21 @@ Saves user's changes as a [style variation](https://developer.wordpress.org/them
--   Embed Google font in your active theme
--   Embed local font file assets
 
 ### Step 2: Fonts, styles and templates customizations
```
```suggestion
### Step 2: Styles and templates customizations
```

This isn't really incorrect, but I don't think we need to mention fonts in this title anymore.

vcanales committed:
Update README.md

Co-authored-by: Sarah Norris <1645628+mikachan@users.noreply.github.com>

mikachan reviewed (approved):
Looks good!
